### ISSUE TITLE
Query syntax

### ISSUE THREAD

**Issue by echo6ix** [stashapp/stash#185]
Feature enhancement request to add basic query syntax to searches.

Query syntax would be an extension of the filter functionality deployed directly in the search boxes. Query syntax could add the ability to use operators (logical, text, string, numerical, and boolean operators) and to query all existing filter keywords (alias, rating, is_missing, fake_tits, age, height, resolution, etc).

#### Considerations
The devs would have a better idea of how to implement this for the end-user than me. Some thoughts to consider:
- Should filter keywords be distinguish them for non-query syntax keywords? Perhaps filter keywords should have the `$` character prepended to them. Eg. `$alias`, `$rating`, `$age`, `$is_missing`
- Autocomplete entries with drop down list: When typing a filter keyword, show list of matching ${filter keywords} as user types.
- Make it simple for users to combine normal searches with query syntax. Would it be a good idea to surround any query syntax with the backtick character. Eg. search for any performer named jane that does not have fake_tits: `` jane `$fake_tits is false` ``

#### Example operators to consider for inclusion

Operators|Type
-----------|-----
`>`|greater than numerical operator
`<`|less than numerical operator
`=`|equal numerical operator
`!`, `not`|not logical operators
`&`, `and`|and logical operators
`\|\|`, `or`|or logical operators
`has`|text operator (eg. `{keyword} has {string}`)

**Comment by malconfig** [comment]
Maybe consider `"` quotation marks instead of backticks, as it is as far as I know more common across international keyboard layouts. It is also what both Google and DuckDuckGo (and probably more search engines I have not used) use for specifying exact searches, so it would make sense for users of those as well.

**Comment by echo6ix** [comment]
Good point, I didn't know that. The reason I suggested back ticks instead of quotation marks is that the latter already seems reserved for searching literal phrases.

Another common method would be using a reserved followed by a colon ala gmail and Google searches and to surround the query syntax with brackets when using multiple values or an operator. 

For example queries,

- Performers containing the string Jane that has _fake_tits_
  - `jane fake_tits:true`
- _Aliases_ containing the string Lana and older than 30 years of _age_
  - `(alias:jane and age:>=30)`
- Performers containing the string Jane, _aliases_ containing the string Lana, older than 30 years of _age_, and doesn't have _fake_tits_
  - `jane (alias:lana and age:>=30 and fake_tits:false)`
